From eba35fec91295a20f70af1e4d6f8b9cc87fa170c Mon Sep 17 00:00:00 2001
From: Rohit Visavalia <rohit.visavalia@xilinx.com>
Date: Tue, 28 Jul 2020 22:09:32 -0700
Subject: [PATCH 1471/1851] staging: xlnxsync: Check valid bit for luma/chroma
 done interrupts

commit 6daed397d7c4dbdedc73c3d644681e65b93eabce from
https://github.com/Xilinx/linux-xlnx.git

As per SyncIP register spec. luma/chroma frame done bits are valid only if
respective valid bit is set.
Without this fix, it was not making any functionality issue but it was not
showing frame done status for buffer 0 of syncIp.

Signed-off-by: Rohit Visavalia <rohit.visavalia@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/staging/xlnxsync/xlnxsync.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/xlnxsync/xlnxsync.c b/drivers/staging/xlnxsync/xlnxsync.c
index cec913334a37..e8920a6500c9 100644
--- a/drivers/staging/xlnxsync/xlnxsync.c
+++ b/drivers/staging/xlnxsync/xlnxsync.c
@@ -979,28 +979,28 @@ static irqreturn_t xlnxsync_irq_handler(int irq, void *data)
 		    chan->ldiff_err || chan->cdiff_err)
 			chan->err_event = true;
 
-		if (val & XLNXSYNC_ISR_PLDONE_MASK) {
+		if (val & XLNXSYNC_ISR_PLVALID_MASK) {
 			i = (val & XLNXSYNC_ISR_PLDONE_MASK) >>
 				XLNXSYNC_ISR_PLDONE_SHIFT;
 
 			chan->l_done[i][XLNXSYNC_PROD] = true;
 		}
 
-		if (val & XLNXSYNC_ISR_PCDONE_MASK) {
+		if (val & XLNXSYNC_ISR_PCVALID_MASK) {
 			i = (val & XLNXSYNC_ISR_PCDONE_MASK) >>
 				XLNXSYNC_ISR_PCDONE_SHIFT;
 
 			chan->c_done[i][XLNXSYNC_PROD] = true;
 		}
 
-		if (val & XLNXSYNC_ISR_CLDONE_MASK) {
+		if (val & XLNXSYNC_ISR_CLVALID_MASK) {
 			i = (val & XLNXSYNC_ISR_CLDONE_MASK) >>
 				XLNXSYNC_ISR_CLDONE_SHIFT;
 
 			chan->l_done[i][XLNXSYNC_CONS] = true;
 		}
 
-		if (val & XLNXSYNC_ISR_CCDONE_MASK) {
+		if (val & XLNXSYNC_ISR_CCVALID_MASK) {
 			i = (val & XLNXSYNC_ISR_CCDONE_MASK) >>
 				XLNXSYNC_ISR_CCDONE_SHIFT;
 
-- 
2.31.1

